src/wallet/gtest/test_wallet.cpp

```diff
@@ -806,6 +806,127 @@ TEST(WalletTests, GetConflictedSaplingNotes) {
+    EXPECT_TRUE(chainActive.Contains(&fakeIndex));
+    EXPECT_EQ(0, chainActive.Height());
+
+    // Simulate SyncTransaction which calls AddToWalletIfInvolvingMe
+    auto orchardTxMeta = wallet.GetOrchardWallet().AddNotesIfInvolvingMe(wtx);
+    ASSERT_TRUE(orchardTxMeta.has_value());
+    EXPECT_FALSE(orchardTxMeta.value().empty());
+    wtx.SetOrchardTxMeta(orchardTxMeta.value());
+    wtx.SetMerkleBranch(block);
+    wallet.LoadWalletTx(wtx);
+
+    // Simulate receiving new block and ChainTip signal
+    wallet.IncrementNoteWitnesses(Params().GetConsensus(),&fakeIndex, &block, sproutTree, saplingTree, true);
+    wallet.UpdateSaplingNullifierNoteMapForBlock(&block);
+
+    // Fetch the Orchard note so we can spend it.
+    std::vector<SproutNoteEntry> sproutEntries;
+    std::vector<SaplingNoteEntry> saplingEntries;
+    std::vector<OrchardNoteMetadata> orchardEntries;
+    wallet.GetFilteredNotes(sproutEntries, saplingEntries, orchardEntries, std::nullopt, -1);
+    EXPECT_EQ(0, sproutEntries.size());
+    EXPECT_EQ(0, saplingEntries.size());
+    EXPECT_EQ(1, orchardEntries.size());
+
+    // Generate another recipient
+    libzcash::diversifier_index_t j2(1);
+    auto recipient2 = ivk.Address(j2);
+
+    // Generate tx to spend note A
+    auto noteToSpend = std::move(wallet.GetOrchardSpendInfo(orchardEntries)[0]);
+    auto builder2 = TransactionBuilder(consensusParams, 2, orchardTree.root());
+    builder2.AddOrchardSpend(std::move(noteToSpend.first), std::move(noteToSpend.second));
+    auto tx2 = builder2.Build().GetTxOrThrow();
+    CWalletTx wtx2 {&wallet, tx2};
+
+    // Generate conflicting tx to spend note A
+    auto noteToSpend2 = std::move(wallet.GetOrchardSpendInfo(orchardEntries)[0]);
+    auto builder3 = TransactionBuilder(consensusParams, 2, orchardTree.root());
+    builder3.AddOrchardSpend(std::move(noteToSpend2.first), std::move(noteToSpend2.second));
+    auto tx3 = builder3.Build().GetTxOrThrow();
+    CWalletTx wtx3 {&wallet, tx3};
+
+    auto hash = wtx.GetHash();
+    auto hash2 = wtx2.GetHash();
+    auto hash3 = wtx3.GetHash();
+
+    // No conflicts for no spends (wtx is currently the only transaction in the wallet)
+    EXPECT_EQ(0, wallet.GetConflicts(hash).size());
+
+    // No conflicts for one spend
+    auto orchardTxMeta2 = wallet.GetOrchardWallet().AddNotesIfInvolvingMe(wtx2);
+    ASSERT_TRUE(orchardTxMeta2.has_value());
+    EXPECT_FALSE(orchardTxMeta2.value().empty());
+    wtx2.SetOrchardTxMeta(orchardTxMeta2.value());
+    wallet.LoadWalletTx(wtx2);
+    EXPECT_EQ(0, wallet.GetConflicts(hash).size());
+
+    // Conflicts for two spends
+    auto orchardTxMeta3 = wallet.GetOrchardWallet().AddNotesIfInvolvingMe(wtx3);
+    ASSERT_TRUE(orchardTxMeta3.has_value());
+    EXPECT_FALSE(orchardTxMeta3.value().empty());
+    wtx3.SetOrchardTxMeta(orchardTxMeta3.value());
+    wallet.LoadWalletTx(wtx3);
+    auto c3 = wallet.GetConflicts(hash);
+    EXPECT_EQ(2, c3.size());
```

Comment:
This should never return any conflicts, because `wtx` does not spend any inputs shared by `wtx2` or `wtx3`. In fact, I am confused as to how CI is passing with this test present; the `EXPECT_EQ(2, c3.size());` should be failing. What I expected here is:
```suggestion
    auto c3 = wallet.GetConflicts(hash2);
    EXPECT_EQ(2, c3.size());
```